packages/models/src/lib/plugin-config.unit.test.ts

```diff
@@ -1,101 +1,109 @@
 import { describe, expect, it } from 'vitest';
-import { config, pluginConfig } from '../../test';
-import { AuditOutputs } from './audit-output';
-import { pluginConfigSchema } from './plugin-config';
+import { PluginConfig, pluginConfigSchema } from './plugin-config';
 
 describe('pluginConfigSchema', () => {
-  it('should parse if plugin configuration is valid', () => {
-    const pluginConfig = config().plugins[0];
-    expect(() => pluginConfigSchema.parse(pluginConfig)).not.toThrow();
+  it('should accept a valid plugin configuration with all entities', () => {
+    expect(() =>
+      pluginConfigSchema.parse({
+        slug: 'eslint',
+        title: 'ESLint plugin',
+        description: 'This plugin checks ESLint in configured files.',
+        docsUrl: 'https://eslint.org/',
+        icon: 'eslint',
+        runner: { command: 'node', outputFile: 'output.json' },
+        audits: [
+          { slug: 'no-magic-numbers', title: 'Use defined constants' },
+          { slug: 'require-await', title: 'Every async has await.' },
+        ],
+        groups: [
+          {
+            slug: 'typescript-eslint',
+            title: 'TypeScript ESLint rules',
+            refs: [{ slug: 'require-await', weight: 2 }],
+          },
+        ],
+        packageName: 'cli',
+        version: 'v0.5.2',
+      } satisfies PluginConfig),
+    ).not.toThrow();
   });
 
-  it('should throw if plugin slug has a invalid pattern', () => {
-    const invalidPluginSlug = '-invalid-plugin-slug';
-    const pluginConfig = config().plugins[0];
-    pluginConfig.slug = invalidPluginSlug;
-
-    expect(() => pluginConfigSchema.parse(pluginConfig)).toThrow(
-      `slug has to follow the pattern`,
-    );
+  it('should accept a minimal plugin configuration', () => {
+    expect(() =>
+      pluginConfigSchema.parse({
+        slug: 'cypress',
+        title: 'Cypress testing',
+        icon: 'cypress',
+        runner: { command: 'npx cypress run', outputFile: 'e2e-output.json' },
+        audits: [{ slug: 'cypress-e2e', title: 'Cypress E2E results' }],
+      } satisfies PluginConfig),
+    ).not.toThrow();
   });
 
-  it('should throw if plugin audits contain invalid slugs', () => {
-    const invalidAuditRef = '-invalid-audit-slug';
-    const pluginConfig = config().plugins[0];
-    pluginConfig.audits[0].slug = invalidAuditRef;
-
-    expect(() => pluginConfigSchema.parse(pluginConfig)).toThrow(
-      `slug has to follow the patter`,
-    );
+  it('should accept a plugin configuration with no audits or groups', () => {
```

Comment:
It's questionable if this is correct, a plugin without audits doesn't measure anything (only groups are optional).

Out of scope of this PR, though.

Reply:
#406 created